### PR TITLE
refactor(broker): use raw subscribe in subscriber plugin

### PR DESCRIPTION
**Pull request by harbu**
## Summary

Use new client feature raw subscription 482e471fb1823fc0558fb9cdd271d64e06605f8d to simplify `SubscriberPlugin`. Notice deprecation of `subscriptionRetryInterval`.

Partly fixes NET-954.

## Limitations and future improvements

`StreamrClient` subscriptions should stay in effect until unsubscribe or the client is destroyed. This is why I removed the subscribe interval logic. There is a slight risk that a subscription could be lost earlier if we have a bug in client. But I think "dogfooding" here makes sense since such a (theoretical) bug would _should_ be catched and fixed!

## Checklist before requesting a review

- [x] Is this a breaking change? If it is, be clear in summary.
- [x] Read through code myself one more time.
- [x] Make sure any and all `TODO` comments left behind are meant to be left in.
- [x] Has reasonable passing test coverage?
- [x] Updated changelog if applicable.
- [x] Updated documentation if applicable.

### CODE REVIEW

File: packages/broker/src/plugins/subscriber/SubscriberPlugin.ts [hunk 1]
```diff
@@ -9,60 +10,39 @@ interface ConfigStream {
...
-        this.subscriptionIntervalRef = setTimeout(() => this.subscriptionIntervalFn(), this.subscriptionRetryInterval)
+        this.subscriptions = await Promise.all(this.streamParts.map(({ id, partition }) => (
+            this.streamrClient.subscribe({ id, partition, raw: true })
+        )))
     }
 
     async start(): Promise<void> {
         await this.subscribeToStreamParts()
-        this.subscriptionIntervalRef = setTimeout(() => this.subscriptionIntervalFn(), this.subscriptionRetryInterval)
         logger.info('Started subscriber plugin')
```
Comment (teogeb) [inline packages/broker/src/plugins/subscriber/SubscriberPlugin.ts:27]: Could move this logging to the start of the method, and use the same wording as in the `stop` method (i.e. `Start subscriber plugin`).

File: packages/broker/src/plugins/subscriber/SubscriberPlugin.ts [hunk 2]
```diff
@@ -9,60 +10,39 @@ interface ConfigStream {
...
-    private readonly subscriptionRetryInterval: number
-    private subscriptionIntervalRef: NodeJS.Timeout | null
+    private readonly streamParts: { id: string, partition: number }[]
+    private subscriptions: Subscription[] = []
 
     constructor(options: PluginOptions) {
         super(options)
         this.streamParts = this.pluginConfig.streams.map((stream) => {
-            return toStreamPartID(toStreamID(stream.streamId), stream.streamPartition)
+            const streamPart = toStreamPartID(toStreamID(stream.streamId), stream.streamPartition)
```
Comment (teogeb) [inline packages/broker/src/plugins/subscriber/SubscriberPlugin.ts]: Do we actually need this "transformation" to type safe streamId + partition? We could use the config array directly in the `this.streamrClient.subscribe` call. That way the the values are validated by the client immediately when plugin's `start()` method is called.

**Review by teogeb — CHANGES_REQUESTED** [review]
Could add a changelog entry and/or automatic migration which removed `subscriptionRetryInterval` config option?

> StreamrClient subscriptions should stay in effect until unsubscribe or the client is destroyed. This is why I removed the subscribe interval logic.

This is a great improvement! 👍

File: packages/utils/src/allOrCleanup.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+/**
+ * Given an iterable of promises, settles them one at a time.
+ * If all of them resolve, returns the list of values.
+ * If one of them rejects, provides a callback for cleaning up the
+ * preceding (already resolved) values.
+ */
+export async function allOrCleanup<T>(
```
Comment (teogeb) [inline packages/utils/src/allOrCleanup.ts]: Nice utility 👍 Could consider if we want to name this to e.g. to `promiseTransaction`? It either succeeds or fails, just like a DB transaction, and can be rolled back.

File: packages/broker/src/plugins/subscriber/SubscriberPlugin.ts (hunk 2 above)
Comment (harbu) [reply packages/broker/src/plugins/subscriber/SubscriberPlugin.ts]: Removed.

File: packages/utils/src/allOrCleanup.ts (hunk 3 above)
Comment (harbu) [reply packages/utils/src/allOrCleanup.ts]: Renamed.

File: packages/broker/src/plugins/subscriber/SubscriberPlugin.ts (hunk 1 above)
Comment (harbu) [reply packages/broker/src/plugins/subscriber/SubscriberPlugin.ts:27]: Not going to do in this PR. Reason: naming convention with rest of plugins.

**Pull request merged**